reports-api/src/reports_api/services/event_template.py

```diff
@@ -65,36 +70,125 @@ def import_events_template_new(cls, configuration_file):
+            template_ids = list(map(lambda x: x.id, existing_events))
+            existing_outcomes = OutcomeTemplate.find_by_template_ids(template_ids)
+            outcome_ids = list(map(lambda x: x.id, existing_outcomes))
+            existing_actions = ActionTemplate.find_by_outcome_ids(outcome_ids)
             for event in parent_events:
                 event['phase_id'] = phase_result.id
                 event['start_at'] = str(event['start_at'])
                 event_result = cls._save_event_template(existing_events, event, phase_result.id)
-                (phase_result_copy['events']).append(res.EventTemplateResponseSchema().dump(event_result))
-                child_events = list(filter(lambda x, _parent_id=event['no']: 'parent_id'
-                                           in x and x['parent_id'] == _parent_id, event_dict))
+                child_events = copy.deepcopy(list(filter(lambda x, _parent_id=event['no']: 'parent_id'
+                                                         in x and x['parent_id'] == _parent_id, event_dict)))
+                outcome_dict = outcome_dict.replace({'event_template_id': event["no"]},
+                                                    {'event_template_id': event_result.id}, regex=True)
+                outcome_results = cls._handle_outcomes(outcome_dict, existing_outcomes,
+                                                       existing_actions, action_dict, event)
+                event_result_copy = res.EventTemplateResponseSchema().dump(event_result)
+                event_result_copy = res.EventTemplateResponseSchema().dump(event_result_copy)
+                event_result_copy['outcomes'] = outcome_results
+                (phase_result_copy['events']).append(event_result_copy)
                 for child in child_events:
                     child['phase_id'] = phase_result.id
                     child['parent_id'] = event_result.id
                     child['start_at'] = str(child['start_at'])
                     child_event_result = cls._save_event_template(existing_events, child,
                                                                   phase_result.id, event_result.id)
-                    (phase_result_copy['events']).append(res.EventTemplateResponseSchema().dump(child_event_result))
+                    outcome_dict = outcome_dict.replace({'event_template_id': child["no"]},
+                                                        {'event_template_id': event_result.id}, regex=True)
+                    outcome_results = cls._handle_outcomes(outcome_dict, existing_outcomes,
+                                                           existing_actions, action_dict, child)
+                    event_result_copy = res.EventTemplateResponseSchema().dump(child_event_result)
+                    event_result_copy = res.EventTemplateResponseSchema().dump(event_result_copy)
+                    event_result_copy['outcomes'] = outcome_results

```

Comment:
Same as the above comment. Also, using same variable names in both inner and outer loops could end up replacing the variable in outer loop here.